packages/paste-core/components/combobox/__tests__/multiselect-combobox-customization.spec copy.tsx

```diff
@@ -0,0 +1,206 @@
+    initialSelectedItems={[items[0], items[2]]}
+    groupItemsBy="letter"
+    labelText="Choose a letter:"
+    helpText="This is the help text"
+    initialIsOpen
+    element={element}
+    optionTemplate={(item) => <div>{item.number}</div>}
+    itemToString={(item) => (item ? item.letter : '')}
+    insertBefore={<Text as="span">Z</Text>}
+    insertAfter={<InformationIcon decorative={false} size="sizeIcon20" title="Get more info" />}
+  />
+);
+
+describe('Combobox data-paste-element attributes', () => {
+  it('should set the correct attributes on all MultiselectCombobox components', () => {
+    const {container} = render(<MultiselectComboboxToTest />, {wrapper: initCustomizationWrapper()});
+    expect(container.querySelector('[data-paste-element="MULTISELECT_COMBOBOX"]')).toBeInTheDocument();
+    expect(container.querySelector('[data-paste-element="MULTISELECT_COMBOBOX_WRAPPER"]')).toBeInTheDocument();
+    expect(container.querySelector('[data-paste-element="MULTISELECT_COMBOBOX_ELEMENT"]')).toBeInTheDocument();
+    expect(container.querySelector('[data-paste-element="MULTISELECT_COMBOBOX_CHEVRON_WRAPPER"]')).toBeInTheDocument();
+
+    expect(container.querySelectorAll(`[data-paste-element="MULTISELECT_COMBOBOX_PILL"]`).length).toEqual(2);
+    expect(
+      container.querySelectorAll(`[data-paste-element="MULTISELECT_COMBOBOX_PILL"]`)[0].getAttribute('role')
+    ).toEqual('option');
+
+    expect(container.querySelectorAll(`[data-paste-element="MULTISELECT_COMBOBOX_LIST_ITEM"]`).length).toEqual(3);
+    expect(
+      container.querySelectorAll(`[data-paste-element="MULTISELECT_COMBOBOX_LIST_ITEM"]`)[0].getAttribute('role')
+    ).toEqual('option');
+
+    const [pillgroupListbox, comboboxListbox] = screen.getAllByRole('listbox');
+    expect(pillgroupListbox.getAttribute('data-paste-element')).toEqual('MULTISELECT_COMBOBOX_PILL_GROUP');
+    expect(comboboxListbox.getAttribute('data-paste-element')).toEqual('MULTISELECT_COMBOBOX_LISTBOX');
+    expect(screen.getAllByRole('group')[0].getAttribute('data-paste-element')).toEqual('MULTISELECT_COMBOBOX_LIST');
+    expect(screen.getAllByRole('presentation')).toHaveLength(2);
+    expect(screen.getAllByRole('presentation')[0].getAttribute('data-paste-element')).toEqual(
+      'MULTISELECT_COMBOBOX_GROUPNAME'
+    );
+
+    expect(container.querySelectorAll('[data-paste-element="MULTISELECT_COMBOBOX_LIST_ITEM_TEXT"]')).toHaveLength(3);
+    expect(container.querySelector('[data-paste-element="MULTISELECT_COMBOBOX_PREFIX"]')).toBeInTheDocument();
+    expect(container.querySelector('[data-paste-element="MULTISELECT_COMBOBOX_SUFFIX"]')).toBeInTheDocument();
+  });
+  it('should set the correct attributes on all MultiselectCombobox components with custom element prop', () => {
+    const {container} = render(<MultiselectComboboxToTest element="FOO" />, {wrapper: initCustomizationWrapper()});
+    expect(container.querySelector('[data-paste-element="FOO"]')).toBeInTheDocument();
```

Comment:
Question: just wanted some clarity on this... I thought we don't usually just query the dom to see if an element with the data-paste-elemnet attribute exists because we want to make sure that it is on the correct element. What is the right way?

Ex. `expect(screen.getByRole('textbox').dataset.pasteElement).toEqual("FOO_INPUT")` instead of `expect(container.querySelector('[data-paste-element="FOO_INPUT"]').toBeInTheDocument()`

Reply:
Your way is better. I was just copying Combobox without thinking about it too deeply.